tic_tac_toe/game.py

```diff
@@ -87,18 +147,60 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    #create board object
+    board = game['board']
+    #check illegal moves
+    if game['winner'] or _board_is_full(board):
+        raise InvalidMovement('Game is over.')
+    if player != game['next_turn']:
+        raise InvalidMovement('"{}" moves next.'.format(game['next_turn']))
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    if not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+    #make actual move
+    board[position[0]][position[1]] = player
+    #set winner
+    winner = _check_winning_combinations(board, player)
+    #explore options if a winner exists
+    if winner:
+        game['winner'] = winner
+        game['next_turn'] = None
+        raise GameOver('"{}" wins!'.format(winner))
+    elif _board_is_full(board):
+        game['next_turn'] = None
+        raise GameOver('Game is tied!'.format(winner))
+    else:
+        game['next_turn'] = game['player1'] if game['next_turn'] == game['player2'] else game['player2']
+    
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    
+    board_string = """
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+"""
+    
+    current_board = game['board']
+    
+    #close, still error
+    return board_string.format(current_board[0][0],current_board[0][1],current_board[0][2],current_board[1][0],current_board[1][1],current_board[1][2],current_board[2][0],current_board[2][1],current_board[2][2])
```

Comment:
You could simplify this using `chain` from the `itertools` built-in package.
```
from itertools import chain
return board_string.format(*chain.from_iterable(game['board']))
```
This will flatten your two-dimensional list and then expand it so it can fill your formatting.